cla_backend/apps/cla_butler/qs_to_file.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+import csv
+from datetime import datetime
+import json
+import logging
+import os
+import re
+import time
+
+from django.db import IntegrityError
+from django.db.models import ForeignKey
+from django.core import serializers
+from django.core.serializers.json import DjangoJSONEncoder
+from django.utils.dateparse import parse_datetime
+
+from jsonfield import JSONField
+
+from cla_common.money_interval.fields import MoneyIntervalField
+from cla_common.money_interval.models import MoneyInterval
+
+
+logger = logging.getLogger(__name__)
+
+
+RE_DATE = re.compile(r'(\d{4})-(\d\d?)-(\d\d?)$')
+RE_DATETIME = re.compile(r'(\d{4})-(\d\d?)-(\d\d?) (\d\d?):(\d\d?):(\d\d?)\.(\d{6})\+(\d\d?):(\d\d?)$')
+WRITE_MODE = 'wb'
+APPEND_MODE = 'a'
+
+
+class QuerysetToFile(object):
+    def __init__(self, path):
+        if not os.path.exists(path):
+            os.makedirs(path)
+        self.path = path
+
+    def get_file_path(self, model, ext='csv'):
+        return os.path.join(self.path, '%s.%s' % (model.__name__, ext))
+
+    def get_name(self, field):
+        field_name = field.name
+        if isinstance(field, ForeignKey):
+            field_name = '%s_id' % field_name
+        return field_name
+
+    def get_value(self, instance, field):
+        val = getattr(instance, self.get_name(field))
+        if isinstance(val, MoneyInterval):
+            val = json.dumps(val.as_dict())
+        if hasattr(val, 'pk'):
+            val = val.pk
+        if val is None:
+            val = ''
+        try:
+            return unicode(val).encode('utf-8')
+        except UnicodeDecodeError:
+            return val
+
+    def dump_to_csv(self, qs):
```

Comment:
The three methods, `dump_to_csv`, `dump_to_yaml` and `dump_values` (which should be called `dump_to_json`), are all very similar and suggest a couple of refactorings.
1. Refactor `dump_to_csv` to extract the CSV writing code into a class or function that allows a similar pattern to the other methods, eg:
```python
with open(file_path, write_mode) as csvfile:
    csv.write(serialize_to_csv(qs))
```
2. Refactor the three methods into a single one which takes the serialization format/serialization function as a parameter, eg:
```python
def dump(self, qs, serialize_fn):
    ...
    with open(file_path, write_mode) as dump_file:
        dump_file.write(serialize_fn(qs))
    ...
```